app/components/Views/confirmations/components/Confirm/Footer/Footer.test.tsx

```diff
@@ -24,19 +27,45 @@ describe('Footer', () => {
     expect(getAllByRole('button')).toHaveLength(2);
   });
 
-  it('should call onConfirm when confirm button is clicked', async () => {
+  it('should call onConfirm when confirm button is clicked', () => {
     const { getByText } = renderWithProvider(<Footer />, {
       state: personalSignatureConfirmationState,
     });
     fireEvent.press(getByText('Confirm'));
     expect(mockConfirmSpy).toHaveBeenCalledTimes(1);
   });
 
-  it('should call onReject when reject button is clicked', async () => {
+  it('should call onReject when reject button is clicked', () => {
     const { getByText } = renderWithProvider(<Footer />, {
       state: personalSignatureConfirmationState,
     });
     fireEvent.press(getByText('Reject'));
     expect(mockRejectSpy).toHaveBeenCalledTimes(1);
   });
+
+  it('render confirm button text "Get Signature" if QR signing is in progress', () => {
+    jest
+      .spyOn(QRHardwareHook, 'useQRHardwareContext')
+      .mockReturnValue({
+        isQRSigningInProgress: true,
+      } as unknown as QRHardwareHook.QRHardwareContextType);
+    const { getByText } = renderWithProvider(<Footer />, {
+      state: personalSignatureConfirmationState,
+    });
+    expect(getByText('Get Signature')).toBeDefined();
```

Comment:
```suggestion
    expect(getByText('Get Signature')).toBeTruthy();
```

the current test will pass even if 'Get Signature' is not found. getByText may return null which would pass as defined in this case. We will need to update the other instances in this test file

Reply:
I updated it here, we can create a separate task to refactor rest of it.